src/main/java/kh/farrukh/springactional/student/StudentServiceImpl.java

```diff
@@ -39,4 +42,21 @@ public void createStudent(StudentCreateRequestDto requestDto) {
     studentRepository.save(student);
   }
 
+  @Override
+  public Student findStudent(Long id) {
+    return studentRepository.findById(id)
+        .orElseThrow(() -> new RuntimeException("student.not_found"));
+  }
+
+  @Override
+  public StudentExamResponseDto examineStudent(Long id) {
+    var student = findStudent(id);
+    var result = getRandomResult();
+    return new StudentExamResponseDto(student.getName(), result);
```

Comment:
You must save exam results somewhere. Implement 'exam_log' table

Reply:
Done